libs/facilities/frontend/view/src/lib/components/detail/detail.page.ts

```diff
@@ -0,0 +1,32 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { IxModule } from '@siemens/ix-angular';
+
+import { facilities } from '../facility';
+
+@Component({
+	selector: 'lib-detail',
+	standalone: true,
+	imports: [CommonModule, IxModule],
+	templateUrl: './detail.page.html',
+	styleUrl: './detail.page.scss',
+	encapsulation: ViewEncapsulation.None,
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class XdDetailPage {
+	facility = this.getFacility();
+
+	constructor(private route: ActivatedRoute) {}
+
+	getFacility() {
+		const facility = facilities.find(
+			(facility) => facility.id === this.route.snapshot.params['id'],
+		);
+		if (facility === undefined) {
+			throw new Error('Facility not found');
+		} else {
+			return facility;
+		}
+	}
```

Comment:
1. Where is typesafeness? 
2. This whole method would be `static`
3. This is ither an `@Input` or normally you do the following: 1. get overview data 2. get detail data from the api via id => you need to create a `domain` lib with a `infrastructure` folder where you now mock the data 
sth like this https://github.com/KonsumGandalf/rsdp/tree/final-branch/libs/student-submissions/frontend/domain

Reply:
I made the facility and the method of type IFacilityMock. If I make the method static I don't have access to this.route anymore so until we do it with Input or other things it has to stay like that. But we can probably do that as part of the facility detail view issue